BlueprintUI/Sources/Element/ElementContent.swift

```diff
@@ -15,20 +15,43 @@ public struct ElementContent: Measurable {
         self.storage = builder
     }
 
-    public func measure(in constraint: SizeConstraint) -> CGSize {
-        return storage.measure(in: constraint)
+    public func measure(in constraint: SizeConstraint, environment: Environment) -> CGSize {
+        return storage.measure(in: constraint, environment: environment)
     }
 
     public var childCount: Int {
         return storage.childCount
     }
 
-    func performLayout(attributes: LayoutAttributes) -> [(identifier: ElementIdentifier, node: LayoutResultNode)] {
-        return storage.performLayout(attributes: attributes)
+    func performLayout(attributes: LayoutAttributes, environment: Environment) -> [(identifier: ElementIdentifier, node: LayoutResultNode)] {
+        return storage.performLayout(attributes: attributes, environment: environment)
     }
 
 }
 
+extension ElementContent {
+    func measurable(in environment: Environment) -> Measurable {
+        struct EnvironmentMeasurable: Measurable {
```

Comment:
Might want to move this out of the func? Eg, not sure how this type appears in the debugger or stacks readability-wise.

Reply:
Like how it demangles? Not sure. It looks fine in Xcode, though.

We use this pattern in the `measureFunction` init too. Seems like the right amount of encapsulation if it's not a type that's being used anywhere else.

Reply:
Yeah, or in the debugger / crash logs – that's why I've avoided this before, but maybe its better now (probably).